File: sources/com/s8/web/front/carbide/structure/InlineSpacer.java
package com.s8.web.front.carbide.structure;

import com.s8.io.bohr.neon.core.NeBranch;
import com.s8.web.front.HTML_NeNode;


/**
 * 
 * @author pierreconvert
 *
 */
public class InlineSpacer extends HTML_NeNode {
	
	/**
	 * 
	 * @param branch
	 * @param width
	 * @return
	 */
	public static InlineSpacer create(NeBranch branch, double width) {
		InlineSpacer title = new InlineSpacer(branch);
		title.setWidth(width);
		return title;
	}

	
	/**
	 * 
	 * @param branch
	 */
	public InlineSpacer(NeBranch branch) {
		super(branch, "/s8-web-front/carbide/structure/InlineSpacer");
	}
	
	
	/**
	 * 
	 * @param name
	 */
	public void setWidth(double width) {
		vertex.setFloat32Field("width", (float) width);
	}
	
}

File: web-sources/S8-pkgs-ui-carbide/structure/InlineSpacer.js



import { NeObject } from '/S8-core-bohr-neon/NeObject.js';
import { S8WebFront } from '/S8-pkgs-ui-carbide/S8WebFront.js';



/**
 * basics
 */
S8WebFront.CSS_import('/S8-pkgs-ui-carbide/structure/Structure.css');


export class InlineSpacer extends NeObject {

    /**
     * 
     */
    constructor() {
        super();
        this.node = document.createElement("span");
        this.node.classList.add("spacer");
    }

  

    getEnvelope() {
        return this.node;
    }


    /**
     * 
     * @param {Popover} popoverBox 
     */
    S8_set_width(width) {
        this.node.style.width = width+"px";
    }

    
    S8_render() { /* continuous rendering approach... */ }
    S8_dispose() { /* continuous rendering approach... */ }
}